### PR TITLE
keepalived: allow sa_pool creation in snat mode.

### PR DESCRIPTION
**Pull request by andrewhit**

### CODE REVIEW

**Comment by andrewhit** [comment]
It is allowed to configure the LIPs in full-nat mode within keepalived.conf so same to src-nat mode.

File: tools/keepalived/keepalived/check/ipwrapper.c [hunk 1]
```diff
@@ -197,16 +197,20 @@ init_service_vs(virtual_server_t * vs)
 			SET_ALIVE(vs);
 	}
 
-	/*Set local ip address in "FNAT" mode of IPVS */
-	if ((vs->loadbalancing_kind == IP_VS_CONN_F_FULLNAT) && vs->local_addr_gname) { 
+	/* Set local ip address in "FNAT" mode of IPVS */
+	if (vs->local_addr_gname &&
+        (vs->loadbalancing_kind == IP_VS_CONN_F_SNAT ||
+         vs->loadbalancing_kind == IP_VS_CONN_F_FULLNAT)) {
```
Comment (ywc689) [inline tools/keepalived/keepalived/check/ipwrapper.c]: place `IP_VS_CONN_F_FULLNAT` before `IP_VS_CONN_F_SNAT` , for  the latter are of less probability to use than the former.

Comment (andrewhit) [reply tools/keepalived/keepalived/check/ipwrapper.c]: Okay. And I will send the code review against devel branch.

**Comment by andrewhit** [comment]
Replaced with https://github.com/iqiyi/dpvs/pull/357

So I am closing this request.

**Pull request closed without merging**